opaque_keys/edx/locator.py

```diff
@@ -1620,3 +1621,59 @@ def html_id(self) -> str:
         # HTML5 allows ID values to contain any characters at all other than spaces.
         # These key types don't allow spaces either, so no transform is needed.
         return str(self)
+
+
+class LibraryCollectionLocator(CheckFieldMixin, LibraryCollectionKey):
+    """
+    When serialized, these keys look like:
+        lib-collection:org:lib:collection-id
+    """
+    CANONICAL_NAMESPACE = 'lib-collection'
+    KEY_FIELDS = ('org', 'lib', 'usage_id')
+    org: str
+    lib: str
+    lib_key: LibraryLocatorV2
+    usage_id: str
+
+    __slots__ = KEY_FIELDS
+    CHECKED_INIT = False
+
+    # Allow usage IDs to contian unicode characters
+    USAGE_ID_REGEXP = re.compile(r'^[\w\-.]+$', flags=re.UNICODE)
+
+    def __init__(self, lib_key: LibraryLocatorV2, usage_id: str):
+        """
+        Construct a CollectionLocator
+        """
+        if not isinstance(lib_key, LibraryLocatorV2):
+            raise TypeError("lib_key must be a LibraryLocatorV2")
+
+        self._check_key_string_field("usage_id", usage_id, regexp=self.USAGE_ID_REGEXP)
+        super().__init__(
+            org=lib_key.org,
+            lib=lib_key.slug,
```

Comment:
Do we need to store `org` and `lib` as separate fields? I would assume you can just story `lib_key` and create a `@property def org(self): return self.lib_key.org` accessor etc.

Reply:
Updated: https://github.com/openedx/opaque-keys/pull/335/commits/168c373f4adafe7c81f707baef3a971f7404acd7